src/geometry/manifold/manifoldutils.cc

```diff
@@ -115,31 +115,25 @@ std::shared_ptr<ManifoldGeometry> createMutableManifoldFromPolySet(const PolySet
   PolySet ps_tri(3, psq.convexValue());
   PolySetUtils::tessellate_faces(psq, ps_tri);
   
-  CGAL_DoubleMesh m;
+  CGAL_EpickMesh m;
 
   if (ps_tri.is_convex()) {
-    using K = CGAL::Epick;
+    using P = CGAL::Epick::Point_3;
     // Collect point cloud
-    std::vector<K::Point_3> points(points3d.size());
+    std::vector<P> points(points3d.size());
     for (size_t i = 0, n = points3d.size(); i < n; i++) {
-      points[i] = vector_convert<K::Point_3>(points3d[i]);
+      points[i] = vector_convert<P>(points3d[i]);
     }
     if (points.size() <= 3) return make_shared<ManifoldGeometry>();
 
     // Apply hull
-    CGAL::Surface_mesh<CGAL::Point_3<K>> r;
-    CGAL::convex_hull_3(points.begin(), points.end(), r);
-    CGALUtils::copyMesh(r, m);
+    CGAL::convex_hull_3(points.begin(), points.end(), m);
```

Comment:
You can use the hull function in manifold. We don't do much optimization for now but we may do them later.

Reply:
Oh sorry, I probably checked the wrong thing.

Reply:
Hah, good to know it's there, not sure about here (would have to inline this logic in a manifold-specific way) but I'll probably give it a spin where we perform the hull primitive itself.